src/components/Editable.js

```diff
@@ -193,6 +193,8 @@ export const Editable = connect()(({ isEditing, thoughtsRanked, contextChain, sh
     throttledChangeRef.current.flush()
   })
 
+  React.useEffect(() => () => throttledChangeRef.current.flush(), []) // clean up function when component unmounts (flushing throttle change)
```

Comment:
Import `useEffect` at top of file like `useRef` and remove extraneous lambda wrapper:

```js
useEffect(() => throttledChangeRef.current.flush, [])
```

Or should it be...?

```js
useEffect(throttledChangeRef.current.flush, [])
```

Reply:
Agreed! We should go with first option of removing extraneous lamba wrapper.
About the useEffect and clean up function:-
```
useEffect(() => {
// this is a side effect function
return ()=>{
// this is a clean up function that runs on unmount.
}
},[])
```
The function passed to the `useEffect `is a side effect function and clean up function (gets called on unmount) is returned by the side effect.